subsys/zigbee/osif/zb_nrf_platform.c

```diff
@@ -502,13 +502,22 @@ void zb_osif_abort(void)
 
 void zb_reset(zb_uint8_t param)
 {
+	uint8_t reas = (uint8_t)SYS_REBOOT_COLD;
+
 	ZVUNUSED(param);
 
 #ifdef CONFIG_ZIGBEE_LIBRARY_NCP_DEV
-	sys_reboot(SYS_REBOOT_NCP);
-#else /* CONFIG_ZIGBEE_LIBRARY_NCP_DEV */
-	sys_reboot(SYS_REBOOT_COLD);
+	reas = (uint8_t)SYS_REBOOT_NCP;
 #endif /* CONFIG_ZIGBEE_LIBRARY_NCP_DEV */
+
+/* For nRF5340DK sys_reboot() does not set reset reason.
+ * Do it manually in this case - NCP samples require this.
+ */
+#ifdef CONFIG_BOARD_NRF5340DK_NRF5340_CPUAPP
+	nrf_power_gpregret_set(NRF_POWER, reas);
+#endif /* CONFIG_BOARD_NRF5340DK_NRF5340_CPUAPP */
```

Comment:
Does this apply to more 53 boards than the 53 DK ?

Should we ifdef on the SOC instead?

Reply:
Good point, ifdef on SoC makes more sense, will change this to use CONFIG_SOC_NRF5340_CPUAPP

Reply:
Changed to CONFIG_SOC_NRF5340_CPUAPP